relay-server/src/actors/processor.rs

```diff
@@ -2287,14 +2294,21 @@ impl EnvelopeProcessorService {
                 let Some(inner_span) = span.value_mut() else {
                     continue;
                 };
-                inner_span.segment_id = transaction_span.segment_id.clone();
-                inner_span.is_segment = Annotated::new(false);
-                add_span(span);
+                // HACK: filter spans based on module until we figure out grouping.
+                let Some(span_op) = inner_span.op.value() else {
+                    continue;
```

Comment:
The span is cloned even though you continue a couple of times. Could you move the call to `span.clone()` into the `if` branch below? For example:

1. grab `let Some(inner_span) = span.value() else ...`
2. do all checks on the value and the op
3. in the branch call `inner_span.clone()`
4. make your mutations
5. call `add_span(Annotated::new(inner_span))`